### DISCUSSION TITLE
Best practices for server-side background task processing

### DISCUSSION THREAD

**Discussion by lewebsimple** [nuxt/nuxt#17899] [Questions]
Hello!

I want to integrate some kind of background task processing in my Nuxt3 application, probably with a solution like [BullMQ](https://github.com/taskforcesh/bullmq) and I'm wondering what would be the proper way of achieving this.

More specifically, where should I instantiate the various queues ? inside a hook (which hook? in a module?)? in a server-only plugin? directly inside an API route?

Of course, this would require a Node environment... but I'm also interested in serverless solutions for this.

Any help / guidance is welcome!

_Update:_ So I figured out the queue / worker could be instantiated from just about anywhere, as they will be initialized "outside" of the route handler itself anyways. The next challenge will be to have the workers run in sandboxed processes, I'm not sure this is even possible with nitro right now. I'll probably have to compile / transpile the source separately somehow...

**Comment by Aareksio** [comment]
You _could_ hook into the h3 server process running Nuxt in Node.js mode. This can be done with the use of _[hooks](https://v3.nuxtjs.org/docs/advanced/hooks/#nuxt-hooks)_, eg. `listen`.


Should you? I say no, spin a separate app for stateful operations.

**Comment by lewebsimple** [comment]
See this [answer](https://github.com/nuxt/framework/discussions/1788#discussioncomment-1635426) for a possible solution using nitro preset.
This doesn't solve the problem of compiling sandboxed workers though.
